### DISCUSSION TITLE
How build go program with godror in windows after run it in aix

### DISCUSSION THREAD

**Discussion by azamss** [godror/godror#193] [Q&A]
Hi , i used docker in windows for cross compile godror but dont create file for aix. How connect to oracle in windows and build it for running in aix?

**Comment by tgulacsi** [comment]
Is therr a Go compiler on AIX? gccgo may work, but I've never used it.

**Comment by tgulacsi** [comment]
I've left AIX 5.2 with GCC 4.8, without Go.
As far as I see, now Go 1.16.2 is available, so native compilation is possible.

What have you tried?
How did it fail?

**Reply by azamss** [reply]
i used this commands in windows 
 go get github.com/karalabe/xgo
 xgo --targets=aix/ppc64 -go latest github.com/godror/godror

in project i used godror , i want build project and create binary for aix . after run binary in aix but this command dont create godror for aix.

when change targets to linux my godror created.

**Comment by tgulacsi** [comment]
xgo will compile the given package.
Godror does not have a main package, so nothing can be compiled.
Try something that is a main package and imports
github.com/godror/godror - for example github.com/UNO-SOFT/dbcsv/csvdump

    xgo --targets=aix/ppc64 -go latest -v -x github.com/UNO-SOFT/dbcsv/csvdump

BUT according to the README
(https://github.com/karalabe/xgo#limit-build-targets),
xgo does not suppot AIX!

You can try
https://github.com/godror/godror/blob/main/contrib/cross-compile.sh as
Zig seems to support AIX/powerpc64:
https://ziglang.org/documentation/0.8.1/#Targets

But you have to figure out the proper naming...

Tamás

**Reply by tgulacsi** [reply]
First obstacle is that zig calls it powerpc64, Go ppc64. I've fixed it in cross-compile.sh.
The second is that Go sends "-maix64" to zigcc, which uses clang, which does not like this.
Removing such arguments with sed in contrib/zigcc fixes this, but then "stdlib.h" is missing.
I haven't examined this thoroughly, but seems that zig does not include a libc for aix (see "Available libcs" in the documentation).
